### ISSUE TITLE
Problem with q-input masked when trying to emit some events like @keydown.tab

### ISSUE THREAD

**Issue by henriquemachad0** [quasarframework/quasar#11120]
**Describe the bug**
When applying a mask with q-input  the component to emit some events like @keydown.tab, So it is not possible to handle the event when pressing Tab

**Codepen/jsFiddle/Codesandbox (required)**
https://codepen.io/henriquemachad0/pen/YzxVNaK

**To Reproduce**

1. Add a mask to a q-input
2. In q-input add the event by pressing tab (@keydown.tab)
4. Click on q-input and press tab

**Expected behavior**
Emit the keydown.tab event

**Screenshots**
![image](https://user-images.githubusercontent.com/74209949/138764270-d8a7aac5-3a10-48c9-842f-080a132f1814.png)
![image](https://user-images.githubusercontent.com/74209949/138764295-9a002164-3760-4182-b872-f062711efbdb.png)
![image](https://user-images.githubusercontent.com/74209949/138764324-1cd139ea-33ae-49fb-97a9-4dd7c9f42de4.png)
![image](https://user-images.githubusercontent.com/74209949/138764356-bc49fc31-bddc-4246-b3c1-9f3be6543eae.png)

**Platform (please complete the following information):**
Quasar Version: v2.2.1

Quasar mode:
  - [x] SPA
  - [ ] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

Tested on:
  - [x] SPA
  - [ ] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

OS:
Browsers: Chrome

**Comment by mariopries** [comment]
the same problem here

**Comment by ddenev** [comment]
Same problem in v2.10.1

The issue comes from this line in `QInput.js` (line 96 in v2.10.1):
```js
if (hasMask.value === true) {
  evt.onKeydown = onMaskedKeydown
}
```
and in `onMaskedKeydown` in `use-mask.js`:
```js
function onMaskedKeydown (e) {
  if (shouldIgnoreKey(e) === true) {
    return
  }
  const
    inp = inputRef.value,
    start = inp.selectionStart,
    end = inp.selectionEnd

  if (e.keyCode === 37 || e.keyCode === 39) { // Left / Right
    const fn = moveCursor[ (e.keyCode === 39 ? 'right' : 'left') + (props.reverseFillMask === true ? 'Reverse' : '') ]

    e.preventDefault()
    fn(inp, start, end, e.shiftKey)
  }
  else if (
    e.keyCode === 8 // Backspace
    && props.reverseFillMask !== true
    && start === end
  ) {
    moveCursor.left(inp, start, end, true)
  }
  else if (
    e.keyCode === 46 // Delete
    && props.reverseFillMask === true
    && start === end
  ) {
    moveCursor.rightReverse(inp, start, end, true)
  }
}
```

You may notice that `onMaskedKeydown` actually "steals" the `keydown` event handling and never passes to our userland-defined `keydown` event handler.

@pdanpdan @rstoenescu - please excuse me for calling you directly but this issue is currently a critical one for my project. Could you please check if my analysis is correct and prioritize the issue?

Thank you in advance!

**Comment by rstoenescu** [comment]
Fix will be available in Quasar v2.10.2 and v1.22.2

**Comment by ddenev** [comment]
Thank you, @rstoenescu and @pdanpdan!
